### PR TITLE
Parity superpeer

### PR DESCRIPTION
**Pull request by compscidr**
Merging in parity on Kovan network running in a virtualbox-vm using vagrant.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,6 +74,51 @@ dependencies {
...
+task cloneParity(type: com.palominolabs.gradle.task.git.clone.GitCloneTask) {
+  dir = file("$buildDir/parity")
+  uri = 'https://github.com/RightMesh/vagrant-eth-env.git'
+  treeish = 'origin/master' // a commit hash, or tag name, or branch name (with remote prefix, like 'origin/master')
+}
+
+task cloneStatus {
+    doLast {
+        if (cloneParity.state.failure != null) {
+            //here is what shoud be executed if cloneParity fails
```
Comment (ConnorHeidema) [inline build.gradle]: *should

File: build.gradle [hunk 2]
```diff
@@ -15,23 +29,26 @@ apply plugin: 'application'
...
                 username artifactory_app_username
                 password artifactory_app_password
             }
         }
-        mavenCentral()
-        jcenter()
     }
     dependencies {
-        classpath 'io.left.rightmesh:rightmesh-plugin:1.3'
+        classpath 'io.left.rightmesh:rightmesh-plugin:1.4-SNAPSHOT'
```
Comment (ConnorHeidema) [inline build.gradle]: Is this meant to be a SNAPSHOT still?

File: build.gradle [hunk 3]
```diff
@@ -57,6 +74,51 @@ dependencies {
...
+    doLast {
+        if (cloneParity.state.failure != null) {
+            //here is what shoud be executed if cloneParity fails
+            println 'Failed to use ssh to connect to gitlab. Setup your ssh key and test with ssh -T git@github.com'
+        }
+    }
+}
+
+// creates and starts a parity VM
+task startParity(type: VagrantUp) {
```
Comment (ConnorHeidema) [inline build.gradle:95]: Very minor.
On ln93 you have
type: VagrantUp
On ln100 you have
type:VagrantDestroy

Spacing should be consistent. Also the end bracket after VagrantDestroy in ln 100 is not standard

File: build.gradle (hunk 1 above)
Comment (compscidr) [reply build.gradle]: derp.

File: build.gradle (hunk 2 above)
Comment (compscidr) [reply build.gradle]: oh - nope, good catch!

File: build.gradle [hunk 4]
```diff
@@ -57,6 +74,51 @@ dependencies {
     testCompile 'junit:junit:4.12'
 }
 
+task cloneParity(type: com.palominolabs.gradle.task.git.clone.GitCloneTask) {
+  dir = file("$buildDir/parity")
+  uri = 'https://github.com/RightMesh/vagrant-eth-env.git'
+  treeish = 'origin/master' // a commit hash, or tag name, or branch name (with remote prefix, like 'origin/master')
```
Comment (FrazerSeymour) [inline build.gradle]: Does this comment need to be here?

Also - do you always want to be linked to `HEAD` of `origin/master`? Would a fixed commit be safer so it doesn't break on you unexpectedly?

File: build.gradle [hunk 5]
```diff
@@ -57,6 +74,51 @@ dependencies {
...
+}
+
+task cloneStatus {
+    doLast {
+        if (cloneParity.state.failure != null) {
+            //here is what should be executed if cloneParity fails
+            println 'Failed to use ssh to connect to gitlab. Setup your ssh key and test with ssh -T git@github.com'
```
Comment (FrazerSeymour) [inline build.gradle]: Might help to link to documentation: https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/

Also I'm guessing it's GitHub not GitLab?

Comment (compscidr) [reply build.gradle]: Yes this is github, good idea.

File: build.gradle (hunk 4 above)
Comment (compscidr) [reply build.gradle]: Yes that's probably best - tbh I just used the default that already existed for this. Will update to the most recent commit.

**Review by FrazerSeymour — APPROVED** [review]

**Pull request merged**